src/special_arrays/maybe_hot_vector.jl

```diff
@@ -64,7 +64,12 @@ Zygote.@adjoint A::AbstractMatrix * b::MaybeHotVector = (_check_mul(A, b); Zygot
 _mul(A::AbstractMatrix, b::MaybeHotVector{Missing}) = fill(missing, size(A, 1))
 _mul(A::AbstractMatrix, b::MaybeHotVector{<:Integer}) = A[:, b.i]
 
+# this is a bit shady because we're overloading unexported method not intended for public use
+Flux._fast_argmax(x::MaybeHotVector) = x.i
+
 Flux.onehot(x::MaybeHotVector{<:Integer}) = Flux.onehot(x.i, 1:x.l)
+maybecold(x::MaybeHotVector{Missing}, labels = 1:length(x)) = missing
```

Comment:
It's not necessary to construct the range here

```julia
julia> @btime 1:length(y)
  20.157 ns (2 allocations: 48 bytes)
1:1000

julia> y = rand(1000); @btime 1:length(y);
  23.592 ns (2 allocations: 48 bytes)

julia> @btime nothing
  0.036 ns (0 allocations: 0 bytes)
```

Reply:
good catch